pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -198,3 +175,87 @@ func buildRouteDestination() []networking.HTTPRouteDestination {
 		},
 	}
 }
+
+// getRouteWeights generates a list of integer percentages for route weights that sum to 100.
+// If the number of routes does not evenly divide 100, the weights are calculated as follows:
+// Round all the weights down, find the difference between that sum and 100, then distribute
+// the difference among the weights.
+//
+// e.g. if numRoutes = 6, then 100/6 = 16.666, which rounds down to 16, with a remainder of 100 % 6 = 4.
+// The final percentages would be [17, 17, 17, 17, 16, 16].
+func getRouteWeights(numRoutes int) []int {
+	weights := make([]int, numRoutes)
+	uniformRouteWeight := 100 / numRoutes // round down
+	remainder := 100 % numRoutes
+
+	for i := 0; i < numRoutes; i++ {
+		weights[i] = uniformRouteWeight
+		if i < remainder {
+			weights[i]++
+		}
+	}
+
+	return weights
+}
+
+// buildPathMatchers creates regex matchers for a given route path.
+// These matchers are used in the virtual service to determine which path a request was sent to
+func buildPathMatchers(urlPath string) ([]networking.HTTPMatchRequest, error) {
+	var pathMatchers []networking.HTTPMatchRequest
+	urlPath = path.Join("/", urlPath, "/")
+	regexpPath, err := v1alpha1.BuildPathRegexp(urlPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert path to regexp: %s", err)
+	}
+
+	pathMatchers = append(pathMatchers, networking.HTTPMatchRequest{
```

Comment:
I think this could be returned directly rather than appending to the empty `pathMatchers`.

```
return []networking.HTTPMatchRequest{
  {
	URI: &istio.StringMatch{
		Regex: regexpPath,
    },
  },
}, nil
```

Reply:
done